base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
```

Comment:
- [ ] Please define the relationship between the Decorator and Nameable, it has to be done by using the accessor's attributes such as `attr_accessor :nameable`. 🙏

Reply:
Ok. Thanks